be/src/util/arrow/row_batch.cpp

```diff
@@ -172,6 +201,98 @@ class FromRowBatchConverter : public arrow::TypeVisitor {
 
 #undef PRIMITIVE_VISIT
 
+    // process string-transformable field
+    arrow::Status Visit(const arrow::StringType& type) override {
+        arrow::StringBuilder builder(_pool);
+        size_t num_rows = _batch.num_rows();
+        builder.Reserve(num_rows);
+        for (size_t i = 0; i < num_rows; ++i) {
+            auto cell_ptr = _cur_slot_ref->get_slot(_batch.get_row(i));
+            PrimitiveType primitive_type = _cur_slot_ref->type().type;
+            switch (primitive_type) {
+                case TYPE_VARCHAR:
+                case TYPE_CHAR:
+                case TYPE_HLL: {
+                    const StringValue* string_val = (const StringValue*)(cell_ptr);
+                    if (string_val->ptr == NULL) {
+                        if (string_val->len == 0) {
+                            // 0x01 is a magic num, not usefull actually, just for present ""
+                            //char* tmp_val = reinterpret_cast<char*>(0x01);
+                            ARROW_RETURN_NOT_OK(builder.Append(""));        
+                        } else {
+                            ARROW_RETURN_NOT_OK(builder.AppendNull());
+                        }
+                    } else {
+                        ARROW_RETURN_NOT_OK(builder.Append(std::move(string_val->to_string())));
```

Comment:
```suggestion
                        ARROW_RETURN_NOT_OK(builder.Append(string_val->to_string()));
```
There is no need to add std::move here, because string_val->to_string() is already a r-value reference.